arch/arm/core/aarch64/arm_mmu.c

```diff
@@ -0,0 +1,477 @@
+	MMU_REGION_FLAT_ENTRY("zephyr_rodata",
+			      (uintptr_t)_image_rodata_start,
+			      (uintptr_t)_image_rodata_size,
+			      MT_RODATA | MT_SECURE),
+
+	/* Mark rest of the zephyr execution regions (data, bss, noinit, etc.)
+	 * cacheable, read-write
+	 * Note: read-write region is marked execute-ever internally
+	 */
+	MMU_REGION_FLAT_ENTRY("zephyr_data",
+			      (uintptr_t)__kernel_ram_start,
+			      (uintptr_t)__kernel_ram_size,
+			      MT_NORMAL | MT_RW | MT_SECURE),
+};
+
+static void setup_page_tables(void)
+{
+	unsigned int index;
+	const struct arm_mmu_region *region;
+	u64_t max_va = 0, max_pa = 0;
+
+	for (index = 0; index < mmu_config.num_regions; index++) {
+		region = &mmu_config.mmu_regions[index];
+		max_va = MAX(max_va, region->base_va + region->size);
+		max_pa = MAX(max_pa, region->base_pa + region->size);
+	}
+
+	__ASSERT(max_va <= (1ULL << CONFIG_ARM64_VA_BITS),
+		 "Maximum VA not supported\n");
+	__ASSERT(max_pa <= (1ULL << CONFIG_ARM64_PA_BITS),
+		 "Maximum PA not supported\n");
+
+	/* create translation tables for user provided platform regions */
+	for (index = 0; index < mmu_config.num_regions; index++) {
+		region = &mmu_config.mmu_regions[index];
+		if (region->size || region->attrs)
+			init_xlat_tables(region);
+	}
+
+	/* setup translation table for zephyr execution regions */
+	for (index = 0; index < ARRAY_SIZE(mmu_zephyr_regions); index++) {
+		region = &mmu_zephyr_regions[index];
+		if (region->size || region->attrs)
+			init_xlat_tables(region);
+	}
+}
+
+static void enable_mmu_el1(unsigned int flags)
+{
+	ARG_UNUSED(flags);
+	u64_t val;
+
+	/* Set MAIR, TCR and TBBR registers */
+	__asm__ volatile("msr mair_el1, %0"
+			:
+			: "r" (MEMORY_ATTRIBUTES)
+			: "memory", "cc");
+	__asm__ volatile("msr tcr_el1, %0"
+			:
+			: "r" (get_tcr(1))
+			: "memory", "cc");
+	__asm__ volatile("msr ttbr0_el1, %0"
+			:
+			: "r" ((u64_t)base_xlat_table)
+			: "memory", "cc");
+
+	/* Ensure these changes are seen before MMU is enabled */
+	__ISB();
+
+	/* Enable the MMU and data cache */
+	__asm__ volatile("mrs %0, sctlr_el1" : "=r" (val));
+	__asm__ volatile("msr sctlr_el1, %0"
+			:
+			: "r" (val | SCTLR_M_BIT | SCTLR_C_BIT)
+			: "memory", "cc");
+
+	/* Ensure the MMU enable takes effect immediately */
+	__ISB();
+
+	LOG_INF("MMU enabled with dcache\n");
+}
+
+/* ARM MMU Driver Initial Setup */
+
+/*
+ * @brief MMU default configuration
+ *
+ * This function provides the default configuration mechanism for the Memory
+ * Management Unit (MMU).
+ */
+static int arm_mmu_init(struct device *arg)
+{
+	u64_t val;
+	unsigned int idx, flags = 0;
+
+	/* Current MMU code supports only EL1 */
+	__asm__ volatile("mrs %0, CurrentEL" : "=r" (val));
+
+	__ASSERT(GET_EL(val) == MODE_EL1,
+		 "Exception level not EL1, MMU not enabled!\n");
+
+	/* Ensure that MMU is already not enabled */
+	__asm__ volatile("mrs %0, sctlr_el1" : "=r" (val));
+	LOG_DBG("sctlr_el1: %x\n", val);
```

Comment:
`LOG_DBG("sctlr_el1: %llx\n", val);`

Reply:
https://gist.githubusercontent.com/stephanosio/4e1c2bd72652c4ae3f806f82253ee94d/raw/7db91923b0a402309e176dd007a8cfcf0deac366/zephyrproject-rtos-zephyr-63051.1.log

Reply:
Thanks for pointing out. Fixed.